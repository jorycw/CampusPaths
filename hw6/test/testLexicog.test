# Simple test of breadth-first search algorithm.
# Verifies that it when finding path it orders by
# characters and then comics books, both lexicographically

CreateGraph characters

AddNode characters A
AddNode characters B
AddNode characters C
AddNode characters D

AddEdge characters A B ba
AddEdge characters A C aa
AddEdge characters B D aa
AddEdge characters C D aa

AddNode characters E
AddNode characters F
AddNode characters G
AddNode characters H

AddEdge characters E G aa
AddEdge characters E F ba
AddEdge characters G H aa
AddEdge characters F H aa

FindPath characters A D

FindPath characters E H

AddNode characters I
AddNode characters J
AddNode characters K

AddEdge characters I J a
AddEdge characters I J b

AddEdge characters J K b
AddEdge characters J K a

FindPath characters I J
FindPath characters J K