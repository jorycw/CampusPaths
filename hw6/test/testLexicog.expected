# Simple test of breadth-first search algorithm.
# Verifies that it when finding path it orders by
# characters and then comics books, both lexicographically

created graph characters

added node A to characters
added node B to characters
added node C to characters
added node D to characters

added edge ba from A to B in characters
added edge aa from A to C in characters
added edge aa from B to D in characters
added edge aa from C to D in characters

added node E to characters
added node F to characters
added node G to characters
added node H to characters

added edge aa from E to G in characters
added edge ba from E to F in characters
added edge aa from G to H in characters
added edge aa from F to H in characters

path from A to D:
A to B via ba
B to D via aa

path from E to H:
E to F via ba
F to H via aa

added node I to characters
added node J to characters
added node K to characters

added edge a from I to J in characters
added edge b from I to J in characters

added edge b from J to K in characters
added edge a from J to K in characters

path from I to J:
I to J via a
path from J to K:
J to K via a